taxi/tests/test_views.py

```diff
@@ -0,0 +1,240 @@
+from django.contrib.auth import get_user_model
+from django.test import TestCase
+from django.urls import reverse
+
+from taxi.models import Manufacturer, Driver, Car
+
+MANUFACTURER_URL = reverse("taxi:manufacturer-list")
+CAR_URL = reverse("taxi:car-list")
+DRIVER_URL = reverse("taxi:driver-list")
+
+
+class PublicViewsTest(TestCase):
+    def test_login_required(self):
+        for url in [MANUFACTURER_URL, CAR_URL, DRIVER_URL]:
+            res = self.client.get(url)
+            self.assertNotEqual(res.status_code, 200)
+
+
+class PrivateViewsTest(TestCase):
+    def setUp(self) -> None:
+        self.user = get_user_model().objects.create_user(
+            username="test",
+            password="<PASSWORD>",
+        )
+        self.client.force_login(self.user)
+
+    def test_retrieve_manufacturers(self):
+        Manufacturer.objects.create(name="VAG")
+        Manufacturer.objects.create(name="BMW")
+        response = self.client.get(MANUFACTURER_URL)
+        self.assertEqual(response.status_code, 200)
+        manufacturers = Manufacturer.objects.all()
+        self.assertEqual(list(response.context["manufacturer_list"]),
+                         list(manufacturers))
+        self.assertTemplateUsed(response,
+                                "taxi/manufacturer_list.html")
+
+    def test_create_manufacturer(self):
+        form_data = {
+            "name": "VAG",
+            "country": "Germany",
+        }
+        self.client.post(reverse("taxi:manufacturer-create"), form_data)
+        manufacturer = Manufacturer.objects.get(name="VAG")
+        self.assertEqual(manufacturer.name, form_data["name"])
+        self.assertEqual(manufacturer.country, form_data["country"])
+
+    def test_retrieve_drivers(self):
+        Driver.objects.create(username="test1",
+                              password="<PASSWORD>",
+                              license_number="BTC12345")
+        Driver.objects.create(username="test2",
+                              password="<PASSWORD>",
+                              license_number="BCC12545")
+        response = self.client.get(DRIVER_URL)
+        self.assertEqual(response.status_code, 200)
+        drivers = Driver.objects.all()
+        self.assertEqual(list(response.context["driver_list"]),
+                         list(drivers))
+        self.assertTemplateUsed(response,
+                                "taxi/driver_list.html")
+
+    def test_create_driver(self):
+        form_data = {
+            "username": "test1",
+            "password1": "1A2g3s4f1234",
+            "password2": "1A2g3s4f1234",
```

Comment:
The passwords used in the test should be strong and meet the application's password policy requirements. Ensure that the password used here is valid according to the application's settings.